packages/tldraw/src/lib/shapes/arrow/components/ArrowTextLabel.tsx

```diff
@@ -54,34 +61,34 @@ export const ArrowTextLabel = React.memo(function ArrowTextLabel({
 				<p style={{ width: width ? width : '9px' }}>
 					{text ? TextHelpers.normalizeTextForDom(text) : ' '}
 				</p>
-				{isEditing && (
-					// Consider replacing with content-editable
-					<textarea
-						ref={rInput}
-						className="tl-text tl-text-input"
-						name="text"
-						tabIndex={-1}
-						autoComplete="off"
-						autoCapitalize="off"
-						autoCorrect="off"
-						autoSave="off"
-						autoFocus
-						placeholder=""
-						spellCheck="true"
-						wrap="off"
-						dir="auto"
-						datatype="wysiwyg"
-						defaultValue={text}
-						onFocus={handleFocus}
-						onChange={handleChange}
-						onKeyDown={handleKeyDown}
-						onBlur={handleBlur}
-						onTouchEnd={stopEventPropagation}
-						onContextMenu={stopEventPropagation}
-						onPointerDown={handleInputPointerDown}
-						onDoubleClick={handleDoubleClick}
-					/>
-				)}
+				<textarea
+					ref={rInput}
+					// We need to add the initial value as the key here because we need this component to
+					// 'reset' when this state changes and grab the latest defaultValue.
+					key={initialText}
```

Comment:
(ditto for other instances of this pattern)

Reply:
that's not a bad idea/point on maybe just using the textarea in general — i like it, it's bold!
we can perhaps maybe lean into the `readonly` property of textareas, if need be.

i'll give it a go 👍

Reply:
I tried just using the textarea and it seems untenable at the moment to remove the elements — it'd require a bigger rethink of things if we went that route.

actually the issue you raised @SomeHats seems to already been in production? it might be a slight variation but it's the same issue at the moment really. so i don't _think_ it's quite a regression for the multiplayer quirk.